python/cudf/cudf/core/dataframe.py

```diff
@@ -793,11 +793,30 @@ def fallback(col, fn):
                 else:
                     result[col] = fallback(rhs._cols[col], _reverse_op(fn))
         elif isinstance(other, Series):
-            raise NotImplementedError(
-                "Series to DataFrame arithmetic not supported "
-                "until strings can be used as indices. Try converting your"
-                " Series into a DataFrame first."
-            )
+            other_cols = other.to_pandas().to_dict()
```

Comment:
I'm concerned about this line, it seems like the most inefficient thing we can do with a `cudf.Series`: copy it to host via Pandas, then expand it into a set of key->value pairs index->value. Does this need to be done more efficiently?

Reply:
other is a cudf.Series here where the index is column names, and the values are scalars. It can only be as large as the dataframe is wide, so we need it on the host to efficiently iterate over the columns.